src/main/java/org/jabref/gui/util/MappedList.java

```diff
@@ -0,0 +1,113 @@
+import java.util.List;
+import java.util.function.Function;
+
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.TransformationList;
+
+/**
+ * MappedList implementation based on https://gist.github.com/TomasMikula/8883719
+ *
+ * @author https://github.com/TomasMikula
+ */
+public final class MappedList<A, B> extends TransformationList<A, B> {
+
+    private final Function<B, A> mapper;
+
+    public MappedList(ObservableList<? extends B> sourceList, Function<B, A> mapper) {
+        super(sourceList);
+        this.mapper = mapper;
+    }
+
+    @Override
+    protected void sourceChanged(ListChangeListener.Change<? extends B> change) {
+        fireChange(new ListChangeListener.Change<A>(this) {
+
+            @Override
+            public boolean wasAdded() {
+                return change.wasAdded();
+            }
+
+            @Override
+            public boolean wasRemoved() {
+                return change.wasRemoved();
+            }
+
+            @Override
+            public boolean wasReplaced() {
+                return change.wasReplaced();
+            }
+
+            @Override
+            public boolean wasUpdated() {
+                return change.wasUpdated();
+            }
+
+            @Override
+            public boolean wasPermutated() {
+                return change.wasPermutated();
+            }
+
+            @Override
+            public int getPermutation(int index) {
+                return change.getPermutation(index);
+            }
+
+            @Override
+            protected int[] getPermutation() {
+                // This method is only called by the superclass methods
+                // wasPermutated() and getPermutation(int), which are
+                // both overridden by this class. There is no other way
+                // this method can be called.
+                throw new AssertionError("Unreachable code");
+            }
+
+            @Override
+            public List<A> getRemoved() {
+                List<A> result = new ArrayList<>(change.getRemovedSize());
+                for (B element : change.getRemoved()) {
+                    result.add(mapper.apply(element));
+                }
+                return result;
+            }
+
+            @Override
+            public int getFrom() {
+                return change.getFrom();
+            }
+
+            @Override
+            public int getTo() {
+                return change.getTo();
+            }
+
+            @Override
+            public boolean next() {
+                return change.next();
+            }
+
+            @Override
+            public void reset() {
+                change.reset();
+            }
+        });
+    }
+
+    @Override
+    public int getSourceIndex(int index) {
+        return index;
+    }
+
+    @Override
+    public A get(int index) {
+        return mapper.apply(super.getSource().get(index));
```

Comment:
This implementation of the MappedList looks identical to the one in the EasyBind package but is different from `MappedList.kt`. 
The mapped list from the corda package runs the `mapper` when the item is [inserted](https://github.com/corda/corda/blob/master/client/jfx/src/main/kotlin/net/corda/client/jfx/utils/MappedList.kt#L19)/updated while this implementation here invokes the `mapper` on every `get`-call. Since the `get` method is called relatively often (e.g. if you scroll in the listview), the EasyBind implementation should only be used for quick mappers. Can you please revert the change and/or write a direct reimplementation of the corda mapped list since we otherwise run again in performance problems with huge groups trees.